python/cudf/cudf/core/column_accessor.py

```diff
@@ -182,7 +182,11 @@ def copy(self, deep=False):
         Make a (shallow) copy of this ColumnAccessor.
         """
         if deep:
-            raise TypeError("Cannot deep copy a ColumnAccessor")
+            return self.__class__(
+                {k: v.copy(deep=True) for k, v in self._data.items()},
+                multiindex=self.multiindex,
+                level_names=self.level_names,
+            )
```

Comment:
since `deep` is a boolean, can we get rid of the `if deep:` check and just make the function return this:
```python
return self.__class__(
    {k: v.copy(deep=deep) for k, v in self._data.items()},
    multiindex=self.multiindex,
    level_names=self.level_names,
)
```

Also, should this do `multiindex=self.multiindex.copy(deep=deep)`?

Reply:
A shallow copy of a columnaccessor should return the same column objects whereas doing deep=deep here will cause shallow copies of all of the column objects erroneously.